pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -100,12 +102,17 @@ class ModelCheckpoint(Callback):
             is saved (``model.save(filepath)``).
         every_n_train_steps: Number of training steps between checkpoints.
             If ``every_n_train_steps == None or every_n_train_steps == 0``, we skip saving during training
-            To disable, set ``every_n_train_steps = 0``. This value must be ``None`` non-negative.
-            This must be mutually exclusive with ``every_n_val_epochs``.
+            To disable, set ``every_n_train_steps = 0``. This value must be ``None`` or non-negative.
+            This must be mutually exclusive with ``train_time_interval`` and ``every_n_val_epochs``.
+        train_time_interval: Checkpoints are monitored at the specified time interval.
+            For all practical purposes, this cannot be smaller than the amount
```

Comment:
if the time interval is smaller than the time to process a single batch then we won't save a checkpoint based on this trigger. we'll always skip because the time between any 2 batches will always be smaller than this interval
```
skip_time = (prev_time_check is None or (now - prev_time_check) < train_time_interval.total_seconds())
```

i dont expect anyone to actually run into this issue, but thought it was worth mentioning as a caveat